File: backend/src/main/java/com/staccato/memory/service/dto/response/MemoryResponse.java
package com.staccato.memory.service.dto.response;

import java.time.LocalDate;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.staccato.memory.domain.Memory;

import io.swagger.v3.oas.annotations.media.Schema;

@Schema(description = "추억 상세 목록 조회 시 각각의 추억 상세에 대한 응답 형식입니다.")
public record MemoryResponse(
        @Schema(example = "1")
        Long memoryId,
        @Schema(example = "https://example.com/memorys/geumohrm.jpg")
        @JsonInclude(JsonInclude.Include.NON_NULL)
        String memoryThumbnailUrl,
        @Schema(example = "런던 추억")
        String memoryTitle,
        @Schema(example = "2024-07-27")
        @JsonInclude(JsonInclude.Include.NON_NULL)
        LocalDate startAt,
        @Schema(example = "2024-07-29")
        @JsonInclude(JsonInclude.Include.NON_NULL)
        LocalDate endAt
) {
    public MemoryResponse(Memory memory) {
        this(
                memory.getId(),
                memory.getThumbnailUrl(),
                memory.getTitle(),
                memory.getStartAt(),
                memory.getEndAt()
        );
    }
}

File: android/Staccato_AN/app/src/main/java/com/on/staccato/data/dto/memory/MemoryResponse.kt
package com.on.staccato.data.dto.memory

import com.on.staccato.data.dto.member.MemberDto
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class MemoryResponse(
    @SerialName("memoryId") val memoryId: Long,
    @SerialName("memoryThumbnailUrl") val memoryThumbnailUrl: String? = null,
    @SerialName("memoryTitle") val memoryTitle: String,
    @SerialName("startAt") val startAt: String? = null,
    @SerialName("endAt") val endAt: String? = null,
    @SerialName("description") val description: String? = null,
    @SerialName("mates") val mates: List<MemberDto>,
    @SerialName("moments") val staccatos: List<MemoryStaccatoDto>,
)
